pkg/rdt/config.go

```diff
@@ -283,16 +284,15 @@ func (a l3PctRangeAllocation) Overlay(baseMask Bitmask) (Bitmask, error) {
 		gap := rdt.info.l3MinCbmBits() - numBits
 
 		// First, widen the mask from the "lsb end"
-		lsbAvailable := lsb - baseMaskLsb
-		if gap <= lsbAvailable {
+		if gap <= lsb {
 			lsb -= gap
 		} else {
-			lsb = baseMaskLsb
+			lsb = 0
 		}
 		// If needed, widen the mask from the "msb end"
 		numBits = msb - lsb + 1
```

Comment:
This is very much subjective, but wouldn't this be a more natural code flow ?
```
...
		gap := rdt.info.l3MinCbmBits() - numBits

		// First, widen the mask from the "lsb end"
		if gap <= lsb {
			lsb -= gap
			gap = 0
		} else {
			gap -= lsb
			lsb = 0
		}
		// If needed, widen the mask from the "msb end"
		msbAvailable := baseMaskNumBits - msb - 1
		if gap <= msbAvailable {
			msb += gap
		} else {
			return 0, rdtError("BUG: not enough bits available for cache bitmask (%s applied on basemask %#x)", a, baseMask)
		}
...
```

Reply:
> This is very much subjective, but wouldn't this be a more natural code flow ?

Haha, I don't think it's subjective. It's just better 😄  Good that someone looks at these with fresh eyes 👍  Now changed.